sqlobject/src/main/java/org/jdbi/v3/sqlobject/SqlObjectFactory.java

```diff
@@ -158,9 +160,18 @@ private Handler buildBaseHandler(Class<?> sqlObjectType, Method method) {
     }
 
     private Handler addDecorators(Handler handler, Class<?> sqlObjectType, Method method) {
-        List<HandlerDecorator> decorators = Stream.of(method.getAnnotations())
-                .map(a -> a.annotationType().getAnnotation(SqlMethodDecoratingAnnotation.class))
-                .filter(Objects::nonNull)
+        List<Class<? extends Annotation>> annotationTypes = Stream.of(method.getAnnotations())
+                .map(Annotation::annotationType)
+                .filter(type -> type.isAnnotationPresent(SqlMethodDecoratingAnnotation.class))
+                .collect(toList());
+
+        DecoratorOrder order = method.getAnnotation(DecoratorOrder.class);
```

Comment:
That's a good idea. I'll make it so you can annotate the sql object type.

There's already issue #438 for considering the annotations on supertypes--let's discuss that part of your suggestion there.